File: Java/src/test/java/org/revian/MyArrayTest.java
package org.revian;

import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

class MyArrayTest {
    private MyArray array;

    @BeforeEach
    void setUp() {
        array = new MyArray(5);
    }

    @Test
    void testAddAndGet() {
        array.add(1);
        assertEquals(1, array.get(0));
    }

    @Test
    void testSize() {
        array.add(1);
        array.add(2);
        assertEquals(2, array.size());
    }

    @Test
    void testAddWhenFull() {
        array.add(1);
        array.add(2);
        array.add(3);
        array.add(4);
        array.add(5);
        assertThrows(IndexOutOfBoundsException.class, () -> array.add(6));
    }

    @Test
    void testGetInvalidIndex() {
        array.add(1);
        assertThrows(IndexOutOfBoundsException.class, () -> array.get(2));
    }
}

File: Scala/src/test/scala/org/revian/MyArrayTest.scala
package org.revian

import org.scalatest.funsuite.AnyFunSuite
import org.scalatest.BeforeAndAfter

class MyArrayTest extends AnyFunSuite with BeforeAndAfter {

  var myArray: MyArray = _

  before {
    myArray = new MyArray(3)
  }

  test("Add element to MyArray") {
    myArray.add(1)
    assert(myArray.size() == 1)
    assert(myArray.get(0) == 1)
  }

  test("Add element beyond capacity") {
    myArray.add(1)
    myArray.add(2)
    myArray.add(3)
    assertThrows[ArrayIndexOutOfBoundsException] {
      myArray.add(4)
    }
  }

  test("Get element at index") {
    myArray.add(10)
    myArray.add(20)
    assert(myArray.get(0) == 10)
    assert(myArray.get(1) == 20)
  }

  test("Get element out of bounds") {
    assertThrows[IndexOutOfBoundsException] {
      myArray.get(3)
    }
  }
}
